packages/server/api/src/app/pieces/piece-metadata-service/index.ts

```diff
@@ -48,3 +49,48 @@ export const getPiecePackage = async (
         }
     }
 }
+
+export function toPieceMetadataModelSummary<T extends PieceMetadataSchema | PieceMetadataModel>(
+    pieceMetadataEntityList: T [],
+    showSuggestedActionsAndTriggers?: boolean,
+): PieceMetadataModelSummary[] {
+    if (!showSuggestedActionsAndTriggers) {
```

Comment:
This function can be replaced with

```ts
export function toPieceMetadataModelSummary<T extends PieceMetadataSchema | PieceMetadataModel>(
    pieceMetadataEntityList: T[],
    showSuggestedActionsAndTriggers?: boolean,
): PieceMetadataModelSummary[] {
    return pieceMetadataEntityList.map((pieceMetadataEntity) => {
        return {
            ...pieceMetadataEntity,
            actions: Object.keys(pieceMetadataEntity.actions).length,
            triggers: Object.keys(pieceMetadataEntity.triggers).length,
            suggestedActions: showSuggestedActionsAndTriggers ? Object.values(pieceMetadataEntity.actions) : undefined,
            suggestedTriggers: showSuggestedActionsAndTriggers ? Object.values(pieceMetadataEntity.triggers) : undefined,
        }
    })
}```

Reply:
I don't want to return full actions and triggers that's why I do the mapping between actions/triggers to {displayName, name}, it's an optimization.